packages/page-addresses/src/modals/Create.tsx

```diff
@@ -30,51 +28,49 @@ interface NameState {
 
 function Create ({ onClose, onStatusChange }: Props): React.ReactElement<Props> {
   const { t } = useTranslation();
-  const { api, isEthereum } = useApi();
+  const { api, isEthereum, systemChain } = useApi();
   const [{ isNameValid, name }, setName] = useState<NameState>({ isNameValid: false, name: '' });
-  const [{ address, addressInput, isAddressExisting, isAddressValid }, setAddress] = useState<AddrState>({ address: '', addressInput: '', isAddressExisting: false, isAddressValid: false, isPublicKey: false });
+  const [{ address, addressInput, isAddressExisting, isAddressValid }, setAddress] = useState<AddrState>({ address: '', addressInput: '', isAddressExisting: false, isAddressValid: false });
+  const latestAddressInput = useRef(address);
   const info = useCall<DeriveAccountInfo>(!!address && isAddressValid && api.derive.accounts.info, [address]);
   const isValid = (isAddressValid && isNameValid) && !!info?.accountId;
 
-  const _onChangeAddress = useCallback(
-    (addressInput: string): void => {
-      let address = '';
-      let isAddressValid = true;
-      let isAddressExisting = false;
-      let isPublicKey = false;
-
-      try {
-        if (isEthereum) {
-          const rawAddress = hexToU8a(addressInput);
+  const _onChangeAddressAsync = useCallback(
+    async (input: string): Promise<void> => {
+      latestAddressInput.current = input;
 
-          address = ethereumEncode(rawAddress);
-          isPublicKey = rawAddress.length === 20;
-        } else {
-          const publicKey = keyring.decodeAddress(addressInput);
+      let address: string | null | undefined = getValidatedAddress(input, isEthereum);
+      let isAddressExisting = false;
 
-          address = keyring.encodeAddress(publicKey);
-          isPublicKey = publicKey.length === 32;
-        }
+      if (!address) {
+        address = await getAddressFromDomain(input, { api, systemChain });
+      }
 
-        if (!isAddressValid) {
-          const old = keyring.getAddress(address);
+      if (address) {
```

Comment:
or should if be `!address`

Reply:
it seems to work correctly -- if the address isn't undefined check keyring if it wasn't already named, if so -- use this name